src/tree/map.rs

```diff
@@ -277,12 +277,7 @@ impl<K, V, C> TreeMap<K, V, C> where C: Compare<K> {
     /// ```
     #[unstable = "matches collection reform specification, waiting for dust to settle"]
     pub fn iter(&self) -> Iter<K, V> {
-        Iter {
-            stack: vec![],
-            node: deref(&self.root),
-            remaining_min: self.length,
-            remaining_max: self.length
-        }
+        Iter { iter: node::Iter::new(node::as_ref(&self.root), self.len) }
```

Comment:
As far as I'm concerned, this is basically "your" module, so whatever you prefer.